### PR TITLE
Interactive To-do List

### PR DESCRIPTION
**Pull request by ehmaddd**
This version of the To-do List is the update of the previous version with CURD operations. But in this update, there are features to select more than one task by using checkboxes. A new file is created and added in index.js as a module that controls checkbox behaviour. When we click any checkbox, it's checked and it's status in the object array as well as localStorage changes to complete and the task is marked as completed by "line-through" crossing it. In the mean time"Clear all completed" button is activated. Using the same, if we have selected tasks using the checkbox, we can delete them all simultaneously. The update will be reflected in the display immediately.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -101,25 +90,55 @@ const addChecktext = () => {
...
+      }
+      for (let j = 0; j < todo.taskArr.length; j += 1) {
+        todo.taskArr[j].index = j + 1;
+      }
+      todo.storeLocalStorage();
+      window.location.reload();
+    });
+  }
+};
+
```
Comment (divinecharlotte) [inline src/index.js]: - [x] kindly consider making sure that the taskArr.filter() method is used on this function to meet this project requirement 
> Implement a function for the "Clear all completed" button (use filter() method).

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @ehmaddd     , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 


- Descriptive PR ✔️
- linters are passing 👍  

## Required Changes ♻️

_Check the comments under the review._



## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (ehmaddd) [reply src/index.js]: I have applied the filter method to update the array as per instructions.

File: src/index.js [hunk 2]
```diff
@@ -92,16 +92,22 @@ const addChecktext = () => {
...
-        todo.taskArr.splice(todo.check[k], 1);
-      }
+      todo.check.sort((a, b) => b - a);
+      resArr = todo.taskArr.filter((task) => {
+        if (task.completed !== 'false') {
+          return false;
+        }
+        return task;
+      });
+      todo.taskArr = resArr;
```
Comment (fobadara) [inline src/index.js:105]: [OPTIONAL] I think could have just made it simpler by doing something like this:

```
todo.taskArr.filter((task) => task.completed === 'true'))
```

**Review by fobadara — APPROVED** [review]
Hi @ehmaddd, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**